compiler/noirc_frontend/src/hir/resolution/path_resolver.rs

```diff
@@ -1,16 +1,20 @@
 use super::import::{resolve_import, ImportDirective, PathResolution, PathResolutionResult};
 use crate::ast::Path;
+use crate::node_interner::DependencyId;
 use std::collections::BTreeMap;
 
 use crate::graph::CrateId;
 use crate::hir::def_map::{CrateDefMap, LocalModuleId, ModuleId};
 
 pub trait PathResolver {
     /// Resolve the given path returning the resolved ModuleDefId.
+    /// If `dependencies` is `Some`, a `DependencyId` for each segment in `path`
+    /// will be pushed.
     fn resolve(
         &self,
         def_maps: &BTreeMap<CrateId, CrateDefMap>,
         path: Path,
+        dependencies: &mut Option<&mut Vec<DependencyId>>,
```

Comment:
Is there a more idiomatic way to do this? This is mainly to avoid allocating a `Vec` unless really needed.